### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by fbordignon** [PMEAL/porespy#705] [Q&A]
Hello everyone, I understand that porespy is distributed under the MIT license, but I've noticed that it requires edt which is GPL. Is it your understanding that porespy is not violating GPL by using edt? Thanks

**Comment by jgostick** [comment]
I hadn't really considered this. I am not super familiar with the nuances of the legal apsects, so could be in the wrong, but we don't actually *distribute* ``edt`` with our code, it is just installed from PyPI and used.  What are your thoughts about this?

**Reply by fbordignon** [reply]
Hey @jgostick thanks for your answer. Well, from what I gather it may not be so safe in legal terms to have this requirement. It may be a bit of a grey area, but I wouldn't risk it. The company I work for is also using edt on a non-GPL project but we are looking for alternatives. I know scipy have similar functionalities but is slower.
We are considering implementing an alternative to edt and publish it on pypi. If we have an update on that I will post it here. Notice that we are a small company and it may take some time. Thanks.

**Reply by jgostick** [reply]
Thanks for bringing this to our attention. As I said, I hadn't really thought about it since we are just academics and make no money from PoreSpy. But I have already reached out to my research office here at UofW to get some expert input. I tried a web search, but the answers were quite convoluted and case-by-case specific.

**Comment by fbordignon** [comment]
Hi @jgostick to update you on the matter of edt
@Arenhart developed [PyEDT](https://github.com/ltracegeo/pyedt) and updated porespy to use it https://github.com/Arenhart/porespy/tree/use_pyedt

So far there has been a slight gain when using pyedt on CPU and a big gain when using it on GPU, check out the benchmark fig on pyedt. Besides, the code can be run on distributed machines/nodes/GPUs. @Arenhart is working on releasing it to pypi and improvements to a new version 0.2.
We've published it as MIT license, so no worries with GPL.
We will use this code as a replacement for edt on porespy internally and release it to our final [GeoSlicer](https://www.ltrace.com.br/geoslicer/) users, so we will be maintaining PyEDT actively. Please, feel free to test it and it would be great if we could integrate to the official porespy

**Reply by ma-sadeghi** [reply]
@fbordignon That's amazing to hear! Have you tested its memory usage as well? Apart from speed, one of the reasons we switched to `edt` was its low memory footprint. If I'm not mistaken `ndimage` uses 3-4x more memory than `edt`.

**Reply by fbordignon** [reply]
Yes, I will let @Arenhart weigh in on that, but I believe we are having more or less the same memory usage of `edt`

**Reply by Arenhart** [reply]
Yes, memory usage is also a main concern for us. It has roughly the same memory requirement of edt, basically it creates an output array and performs calculations inplace. In benchmarks where I stressed the memory use, pyedt would always match or slightly surpass edt on maximum array size.

**Reply by ma-sadeghi** [reply]
@Arenhart In that case, we definitely look forward to the official release of `pyedt` on PyPI and conda-forge, so we can begin the migration. In the meantime, I'll try to install the dev version and test it.
